pkg/kubefed2/delete/type.go

```diff
@@ -14,71 +14,63 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package federate
+package delete
 
 import (
 	"fmt"
 	"io"
 
 	"github.com/golang/glog"
 	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
 
 	apiextv1b1client "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/rest"
 
 	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
 	ctlutil "github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+	"github.com/kubernetes-sigs/federation-v2/pkg/kubefed2/disable"
 	"github.com/kubernetes-sigs/federation-v2/pkg/kubefed2/options"
 	"github.com/kubernetes-sigs/federation-v2/pkg/kubefed2/util"
 )
 
 var (
-	disable_long = `
-		Disables propagation of a Kubernetes API type.  This command
-		can also optionally delete the API resources added by the enable
-		command.
+	type_long = `
+		Deletes API resources used to federate a type. Use
+		"kubefed2 disable" instead if the intention is to only
+		diable propagation of the given type.
 
 		Current context is assumed to be a Kubernetes cluster hosting
 		the federation control plane. Please use the
 		--host-cluster-context flag otherwise.`
 
-	disable_example = `
-		# Disable propagation of the Service type
-		kubefed2 federate disable Service
+	type_example = `
+		# Delete API resources used for propagation of the type Service
+		kubefed2 delete type Service
 
-		# Disable propagation of the Service type and delete API resources
-		kubefed2 federate disable Service --delete-from-api`
+		# Delete API resources used for propagation of resource my-svc of type Service
+		kubefed2 delete resource Service my-svc`
 )
 
-type disableType struct {
+type deleteType struct {
 	options.SubcommandOptions
-	disableTypeOptions
+	deleteTypeOptions
 }
 
-type disableTypeOptions struct {
+type deleteTypeOptions struct {
 	targetName string
-	delete     bool
 }
 
-// Bind adds the join specific arguments to the flagset passed in as an
-// argument.
-func (o *disableTypeOptions) Bind(flags *pflag.FlagSet) {
```

Comment:
I think removal of an explicit flag to delete from the api suggests a check for instances of the federated type before allowing deletion without a `--force` parameter.  The risk of accidentally deleting all instances of a type is considerable otherwise (e.g. `kubefed2 delete type deployment.apps` may remove all FederatedDeployments and remove the target deployments from member clusters).   Given that this problem existed before this PR I'm not suggesting you be responsible for fixing it, but I think the change does suggest a new issue if not implemented here.